lib/github_service/commands/run_test.rb

```diff
@@ -0,0 +1,302 @@
+module GithubService
+  module Commands
+    # = GithubService::Commands::RunTest
+    #
+    # Triggers a build given the configured test repo:
+    #
+    #   Settings.github.run_tests_repo.*
+    #
+    # Which doing so will:
+    #
+    #    - validate the command
+    #    - create a branch
+    #    - create a commit with the travis.yml changes
+    #    - push said branch to the origin
+    #    - create a pull request for that branch
+    #
+    # More info can be found here on how the whole process works:
+    #
+    #   https://github.com/ManageIQ/manageiq-cross_repo-tests
+    #
+    # == Command structure
+    #
+    # @miq-bot run-tests [<repos-to-test>] [including <extra-repos>]
+    #
+    # where:
+    #
+    #   - a `repo` is of the form [org/]repo[@ref|#pr]
+    #   - `repos-to-test` is a list of repos to have tested
+    #   - `extra-repos` is a list of repos (gems) to override in the bundle
+    #
+    # Lists of repos should be comma delimited.
+    #
+    # == Examples
+    #
+    # === In ManageIQ/manageiq PR #1234:
+    #
+    # The following command:
+    #
+    #   @miq-bot run-tests manageiq-ui-classic
+    #
+    # will create a commit with the .travis.yml changes:
+    #
+    #
+    #   @@ -14,6 +14,6 @@ matrix:
+    #      fast_finish: true
+    #    env:
+    #      global:
+    #   -  - REPOS=
+    #   +  - REPOS=ManageIQ/manageiq#12345
+    #      matrix:
+    #   -  - TEST_REPO=
+    #   +  - TEST_REPO=ManageIQ/manageiq-ui-classic
+    #
+    #
+    # === In ManageIQ/manageiq PR #1234
+    #
+    # The following command:
+    #
+    #   @miq-bot run-tests manageiq-api,manageiq-ui-classic \
+    #     including Fryguy/more_core_extensions@feature,Fryguy/linux_admin@feature
+    #
+    # will create a commit with the .travis.yml changes:
+    #
+    #   @@ -14,6 +14,6 @@ matrix:
+    #      fast_finish: true
+    #    env:
+    #      global:
+    #   -  - REPOS=
+    #   +  - REPOS=Fryguy/more_core_extensions@feature,Fryguy/linux_admin@feature,ManageIQ/manageiq#1234
+    #      matrix:
+    #   -  - TEST_REPO=
+    #   +  - TEST_REPO=ManageIQ/manageiq-api
+    #   +  - TEST_REPO=ManageIQ/manageiq-ui-classic
+    #
+    # TODO:  Handle the "self" case, where `manageiq` is also a TEST_REPO
+    #
+    # (maybe include a "self" helper as well?)
+    #
+    # === In ManageIQ/manageiq-ui-classic PR #1234:
+    #
+    # The following command:
+    #
+    #   @miq-bot run-tests manageiq#5678
+    #
+    # will create a commit with the .travis.yml changes:
+    #
+    #   @@ -14,6 +14,6 @@ matrix:
+    #      fast_finish: true
+    #    env:
+    #      global:
+    #   -  - REPOS=
+    #   +  - REPOS=manageiq#5678
+    #      matrix:
+    #   -  - TEST_REPO=
+    #   +  - TEST_REPO=manageiq-ui-classic#1234
```

Comment:
Could we point to the manageiq-cross_repo README for examples instead of duplicating them here?  For the purposes of developers reading the class just an explanation of the ENV vars that need to be set is probably sufficient

Reply:
I actually disagree with you here, because there is an added feature when using the bot of including the PR it is called from is automatically included in the `REPOS` section of the diff.  I think it is worth being clear about that.

I also just wanted the use cases that I was having to use as reference from https://github.com/ManageIQ/miq_bot/issues/446 (and are also out of date) somewhere close when building this so I didn't have to flip back and forth when developing this.

Reply:
That's fair, a sentence stating that the current PR will be added to the TEST_REPO list would probably accomplish the same thing

Reply:
Updated the #446 yml diffs due to changes in the manageiq-cross_repo format (in particular the merging of CORE_REPO and GEM_REPOS into just REPOS)

Reply:
> because there is an added feature when using the bot of including the PR it is called from is automatically included in the REPOS section of the diff. I think it is worth being clear about that.

I agree with documenting that peculiarity, but that wasn't my takeaway from reading these comments.  Like @agrare I also viewed it as a copy/paste from the cross_repo examples, and am also concerned about [Basmati Rice](https://twitter.com/nzkoz/status/538892801941848064) (much like the examples in #446 became Basmati Rice)

> I also just wanted the use cases that I was having to use as reference from #446 (and are also out of date) somewhere close when building this so I didn't have to flip back and forth when developing this.

If it's temporary because this is still WIP and it's convenient, then yeah, this is fine.

Reply:
> Updated the #446 yml diffs due to changes in the manageiq-cross_repo format (in particular the merging of CORE_REPO and GEM_REPOS into just REPOS)

Just to make sure with this as well, @agrare 's comment [here](https://github.com/ManageIQ/miq_bot/issues/446#issuecomment-538412470), while also suffering from the same "out of date" problems as the description, are also covered by the examples in the description now, right?  I was looking at the description trying parse if there was any difference, but didn't see any.

> am also concerned about Basmati Rice ...
>
> If it's temporary because this is still WIP and it's convenient, then yeah, this is fine.

@Fryguy @agrare agreeing with the "Basmati Rice" concern, so maybe we just have one example here as a middle ground (maybe the most complicated one)?

Reply:
Specifically with this one, I trimmed down the comment a bit to just one example, and tried to make it as all inclusive as possible.